ring/token_range.go

```diff
@@ -0,0 +1,136 @@
+package ring
+
+import (
+	"math"
+
+	"github.com/pkg/errors"
+	"golang.org/x/exp/slices" // using exp/slices until moving to go 1.21.
+)
+
+// TokenRanges describes token ranges owned by an instance.
+// It consists of [start, end] pairs, where both start and end are inclusive.
+type TokenRanges []uint32
+
+func (tr TokenRanges) IncludesKey(key uint32) bool {
```

Comment:
I don't know whether it might be of interest for this implementation, but there is a struct called `ringToken` declared in `ownership_priority_queue.go` as:
```
type ringToken struct {
	token     uint32
	prevToken uint32
}
```